### PR TITLE
ref(ui): Refactor `ApiApplications`

### PR DESCRIPTION
**Pull request by billyvg**
* Refactor to typescript
* Remove `grid-emotion`
* Visual design consistency cleanups
* Use async/await for API calls

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/account/apiApplications/details.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import TextCopyInput from 'app/views/settings/components/forms/textCopyInput';
+import apiApplication from 'app/data/forms/apiApplication';
+import getDynamicText from 'app/utils/getDynamicText';
+
+type State = {
+  app: ApiApplication;
+};
+
+class ApiApplicationsDetails extends AsyncView<
+  RouteComponentProps<{}, {}> & AsyncView['props'],
```
Comment (scefali) [inline src/sentry/static/sentry/app/views/settings/account/apiApplications/details.tsx]: @billyvg So this is how you should do route props. I guess I should remove `RouterProps` from our types file and use this instead.

**Review by scefali — APPROVED** [review]

File: src/sentry/static/sentry/app/views/settings/account/apiApplications/details.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/settings/account/apiApplications/details.tsx]: is `<{}, {}>` not the default?

Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/account/apiApplications/details.tsx]: is `<{}, {}>` not the default?

Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/account/apiApplications/details.tsx]: I'd probably prefer to see `RouteComponentProps<{}, {}> & AsyncView['props']` as a `type Props = ...` to keep our components more consistent.

Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/account/apiApplications/details.tsx]: Hmm we seem to inconsistently do both

Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/account/apiApplications/details.tsx]: Nope, it requires args

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/settings/account/apiApplications/details.tsx]: Ah damn

**Pull request merged**